pkg/plugins/policies/meshretry/api/v1alpha1/meshretry.go

```diff
@@ -255,11 +253,12 @@ type Predicate struct {
 	PredicateType PredicateType `json:"predicate"`
 	// Tags is a map of metadata to match against for selecting the omitted hosts. Required if Type is
 	// OmitHostsWithTags
-	Tags map[string]string `json:"tags,omitempty"`
+	Tags map[string]string `json:"tags"`
```

Comment:
Is `PredicateType` supposed to be a discriminator here? I'm not sure, but it seems so based on the doc comments like `Required if Type is OmitHostsWithTags` and `Used for OmitPreviousPriorities`. If it's the case then `tags` and `updateFrequency` probably shouldn't be required. It doesn't affect anything, but I guess it's just cleaner API

Reply:
But it's not mergeable because it's under 

```
HostSelection *[]Predicate `json:"hostSelection,omitempty"`
``` 

right?

Reply:
I'll make Predicate required and Tags and updateFrequency optional.